src/cli/commands/import.js

```diff
@@ -276,12 +307,45 @@ export class Import extends Install {
     this.resolver = new ImportPackageResolver(this.config, this.lockfile);
     this.linker = new PackageLinker(config, this.resolver);
   }
-
+  createLogicalDependencyTree(packageJson: ?string, packageLock: ?string): LogicalDependencyTree {
+    try {
+      if (!packageJson || !packageLock) {
+        throw new Error('files missing');
+      }
+      return new LogicalDependencyTree(packageJson, packageLock);
+    } catch (e) {
+      throw new MessageError(this.reporter.lang('importSourceFilesCorrupted'));
+    }
+  }
+  async getExternalLockfileContents(): Promise<{packageJson: ?string, packageLock: ?string}> {
+    try {
+      const [packageJson, packageLock] = await Promise.all([
+        fs.readFile(path.join(this.config.cwd, NODE_PACKAGE_JSON)),
+        fs.readFile(path.join(this.config.cwd, NPM_LOCK_FILENAME)),
+      ]);
+      return {packageJson, packageLock};
+    } catch (e) {
+      return {packageJson: null, packageLock: null};
+    }
+  }
   async init(): Promise<Array<string>> {
     if (await fs.exists(path.join(this.config.cwd, LOCKFILE_FILENAME))) {
       throw new MessageError(this.reporter.lang('lockfileExists'));
     }
-    await verifyTreeCheck(this.config, this.reporter, {}, []);
+    const {packageJson, packageLock} = await this.getExternalLockfileContents();
+    const importSource =
+      packageJson && packageLock && semver.satisfies(nodeVersion, '>=5.0.0') ? 'package-lock.json' : 'node_modules';
+    if (importSource === 'package-lock.json') {
+      this.reporter.info(this.reporter.lang('importPackageLock'));
+      const tree = this.createLogicalDependencyTree(packageJson, packageLock);
+      if (this.resolver instanceof ImportPackageResolver) {
```

Comment:
Is there a known case where `resolver` isn't `ImportPackageResolver`? If so, what would happen? Since the package-lock tree won't be set on the resolver, should this check maybe be part of `packageJson && packageLock && semver.satisfies(nodeVersion, '>=5.0.0') ? 'package-lock.json' : 'node_modules';` to prevent it from taking this code path? (if resolver is not instanceof ImportPackageResolver then we build the logical tree and don't use it)

Reply:
More `flow` acrobatics, I'm afraid. :/ I've seen this done elsewhere in the file though. I'm not sure if adding that check below will satisfy `flow`, but can give it a shot.

Reply:
Just use a [typecasting expression](https://flow.org/en/docs/types/casting/), or an `invariant` 🙂